SMLHelper/Assets/CustomFishPrefab.cs

```diff
@@ -0,0 +1,132 @@
+﻿namespace SMLHelper.V2.Assets
+{
+    using System;
+    using System.Collections.Generic;
+    using UnityEngine;
+    using SMLHelper.V2.Utility;
+    using SMLHelper.V2.MonoBehaviours;
+
+    /// <summary>
+    /// Class used by CustomFish for constructing a prefab based on the values provided by the user.
+    /// You can use this yourself if you want, but you will need to manually provide a TechType
+    /// </summary>
+    public class CustomFishPrefab : ModPrefab
+    {
+        public GameObject modelPrefab;
+        public float scale;
+        public bool pickupable;
+        public bool isWaterCreature = true;
+
+        public float swimSpeed;
+        public Vector3 swimRadius;
+
+        public List<Type> componentsToAdd = new List<Type>();
+
+        GameObject cachedPrefab;
+
+        public CustomFishPrefab(string classId, string prefabFileName, TechType techType = TechType.None) : base(classId, prefabFileName, techType)
+        {
+        }
+
+        public override GameObject GetGameObject()
+        {
+            if (cachedPrefab)
+            {
+                return cachedPrefab;
```

Comment:
Please explain the benefits of this `cachedPrefab` pattern.  
If it's that good, then maybe this pattern should just be added directly to [ModPrefab](https://github.com/SMLHelper/SMLHelper/blob/2.x/SMLHelper/Assets/ModPrefab.cs) itself.

Reply:
Im using it because a lot of fish are spawned, even during gameplay when the player loads new areas, so this is to reduce the potential performance hit when loading new areas with lots of creatures

Reply:
So there are performance benefits.  
Do you think this could be applied to the `ModPrefab` class directly so that all modded prefabs can get the same benefit?  
Heck, that might even be the solution to the issues we've been seeing where modded prefabs load loaded twice when being deserialized.

Reply:
If you wanna give it a shot, go ahead. It would be very helpful for things with lots of components on them

Reply:
You are also more than welcome to make that change as part of this PR.  
Since you are already making changes anyways, and you clearly understand how this prefab cache pattern works, it seems only right to allow you to make the update.

That way, the `CustomFishPrefab` class wouldn't need this anymore because the base class would handle it instead.

Reply:
Going to test the change for myself before marking this as resolved.